modules/product/src/Event/AvailableVariationsEvent.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce_product\Event\FilterVariationsEvent.
+ */
+
+namespace Drupal\commerce_product\Event;
+
+use Drupal\commerce_product\Entity\Product;
+use Symfony\Component\EventDispatcher\Event;
+
+class AvailableVariationsEvent extends Event {
+
+  /**
+   * The product containing variations to be filtered.
+   *
+   * @var \Drupal\commerce_product\Entity\Product
```

Comment:
Let's use \Drupal\commerce_product\ProductInterface, always better than mentioning a specific class. Same in the constructor docblock and typehint.